src/NHibernate/Util/ReflectHelperMemoryAssembly.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NHibernate.Util
+{
+	public class ReflectHelperMemoryAssembly
+    {
+		private static readonly ReflectHelperMemoryAssembly instance = new ReflectHelperMemoryAssembly();
+		private Dictionary<string, Assembly> _assemblyList;
```

Comment:
A singleton must be thread safe, you should use a [`ConcurrentDictionary<string, Assembly>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2) here, stored in a readonly field.